rust/rbac-registration/src/cardano/cip509/validation.rs

```diff
@@ -273,6 +120,24 @@ pub(crate) fn validate_stake_public_key(
         },
     };
 
+    // TODO: This should be improved, probably along with the `compare_key_hash` function
+    // implementation.
+    let pk_addrs: Vec<_> = cip509
```

Comment:
This needs a comment explaining exactly what its doing, its not possible to read it and know what its doing (or supposed to be doing).  Also looks kind of complex, so its probably a good candidate to be its own function (that explains what its doing) and a unit test which tests that it does what its aiming to do.